internal/services/communication/communication_service_email_domain_association_resource.go

```diff
@@ -0,0 +1,340 @@
+			if err != nil {
+				return fmt.Errorf("parsing Communication Service ID: %w", err)
+			}
+
+			eMailServiceDomainId, err := domains.ParseDomainID(state.EMailServiceDomainId)
+			if err != nil {
+				return fmt.Errorf("parsing EMail Service Domain ID: %w", err)
+			}
+
+			locks.ByName(communicationServiceId.CommunicationServiceName, "azurerm_communication_service")
+			defer locks.UnlockByName(communicationServiceId.CommunicationServiceName, "azurerm_communication_service")
+
+			locks.ByName(eMailServiceDomainId.DomainName, "azurerm_email_communication_service_domain")
+			defer locks.UnlockByName(eMailServiceDomainId.DomainName, "azurerm_email_communication_service_domain")
+
+			existingEMailServiceDomain, err := domainClient.Get(ctx, *eMailServiceDomainId)
+			if err != nil && !response.WasNotFound(existingEMailServiceDomain.HttpResponse) {
+				return fmt.Errorf("checking for the presence of existing EMail Service Domain %q: %+v", state.EMailServiceDomainId, err)
+			}
+
+			if response.WasNotFound(existingEMailServiceDomain.HttpResponse) {
+				return metadata.MarkAsGone(id)
+			}
+
+			existingCommunicationService, err := client.Get(ctx, *communicationServiceId)
+			if err != nil && !response.WasNotFound(existingCommunicationService.HttpResponse) {
+				return fmt.Errorf("checking for the presence of existing Communication Service %q: %+v", state.CommunicationServiceId, err)
+			}
+
+			if response.WasNotFound(existingCommunicationService.HttpResponse) {
+				return metadata.MarkAsGone(id)
+			}
+
+			if existingCommunicationService.Model == nil {
+				return fmt.Errorf("model for %s was nil", state.CommunicationServiceId)
+			}
+
+			if existingCommunicationService.Model.Properties == nil {
+				return fmt.Errorf("properties for %s was nil", state.CommunicationServiceId)
+			}
+
+			domainList := existingCommunicationService.Model.Properties.LinkedDomains
+			if domainList == nil {
+				return fmt.Errorf("checking for Domain Association %s for %s", *eMailServiceDomainId, *communicationServiceId)
+			}
+
+			var found bool
+
+			for _, v := range pointer.From(domainList) {
+				tmpID, tmpErr := domains.ParseDomainIDInsensitively(v)
+				if tmpErr != nil {
+					return fmt.Errorf("parsing domain ID %q from LinkedDomains for %s: %+v", v, communicationServiceId, err)
+				}
+
+				if strings.EqualFold(eMailServiceDomainId.ID(), tmpID.ID()) {
+					found = true
+
+					break
+				}
+			}
+
+			if !found {
+				return metadata.MarkAsGone(id)
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (EmailDomainAssociationResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Communication.ServiceClient
+			domainClient := metadata.Client.Communication.DomainClient
+
+			var model EmailDomainAssociationResourceModel
+
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
```

Comment:
We can only be sure of having the resource ID at this point, so we should not need to do this, and work exclusively with the ID (and the data it contains) for delete operations. It's also unused in the rest of the function here too.
```suggestion
```